src/components/MainSidebar/DimensionsList/DimensionsList.js

```diff
@@ -0,0 +1,111 @@
+import i18n from '@dhis2/d2-i18n'
+import { CircularLoader, NoticeBox } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React, { useRef, useEffect } from 'react'
+import { DimensionListItem } from './DimensionListItem.js'
+import styles from './DimensionsList.module.css'
+
+const getNoResultsMessage = (searchTerm, programName) => {
+    if (searchTerm) {
+        return i18n.t("No dimensions found for '{{searchTerm}}'", {
+            searchTerm,
+        })
+    }
+    if (programName) {
+        i18n.t('No dimensions found in the {{programName}} program', {
+            programName,
+        })
+    }
+    return i18n.t('No dimensions found')
+}
+
+const DimensionsList = ({
+    loading,
+    fetching,
+    error,
+    dimensions,
+    programName,
+    searchTerm,
+    setIsListEndVisible,
+}) => {
+    const scrollBoxRef = useRef()
+    const updateIsListEndVisible = () => {
+        const el = scrollBoxRef.current
+        const isVisible = el.scrollHeight - el.scrollTop - el.clientHeight < 5
+
+        setIsListEndVisible(isVisible)
+    }
+
+    useEffect(() => {
+        if (dimensions && scrollBoxRef.current && !loading && !fetching) {
+            updateIsListEndVisible()
+        }
+    }, [dimensions, scrollBoxRef.current])
+
+    if (loading) {
+        return (
+            <div className={styles.loaderWrap}>
+                <CircularLoader small />
+            </div>
+        )
+    }
+
+    if (error) {
```

Comment:
technically `if` might be correct here, but when reading through a component with multiple `return`s, I think it make sense to wrap them in `if`, `if else` and finally `else` to indicate that they're indeed separate paths.

Reply:
same for the returns in `getNoResultsMessage` above

Reply:
IMO superior to all of this is to avoid multiple returns altogether, assign the value to a `let` in an `if / if else / else` block and use a single return at the end though.

Reply:
This comment threw me a bit. In react components I see `if/return` it much more than `if/else`. I'm happy to comply with the code conventions of the team, but then I think the team is choosing a pattern that's a bit different from the react ecosystem as a whole....

Reply:
I don't think we ever agreed on a team convention. I prefer the `if/return` as well, and if that is more compliant with the react ecosystem, then it makes sense to me to stick with it.

Reply:
I understand both arguments (different paths vs react component convention).

I often think `if / else if` gets messy when the clauses are unrelated.

Reply:
When checking for different values of the same varible then `if / else if` is pretty readable, like `switch / case`.

Reply:
Lets stick to the react component convention in this case.

Reply:
I don't mind the `if/return` if it's a single use (e.g. commonly to check for a prop and return `null` to avoid multiple partial renders while loading). But if it's simply the case of checking for different conditions and returning primitive values (i18n still counts in this case) like it is with `getNoResultsMessage` I don't see why not to use `if / else if / else`.